src/main/webapp/app/shared/profile-toggle/profile-toggle.directive.ts

```diff
@@ -0,0 +1,68 @@
+import { Directive, HostBinding, Input, OnDestroy, OnInit } from '@angular/core';
+import { tap } from 'rxjs/operators';
+import { Subscription } from 'rxjs';
+import { ProfileToggle, ProfileToggleService } from 'app/shared/profile-toggle/profile-toggle.service';
+
+@Directive({
+    selector: '[jhiProfileToggle]',
+})
+export class ProfileToggleDirective implements OnInit, OnDestroy {
+    @Input('jhiProfileToggle') profiles: ProfileToggle | ProfileToggle[];
+    /**
+     * This input must be used to overwrite the disabled state given that the profile toggle is inactive.
+     * If the normal [disabled] directive of Angular would be used, the HostBinding in this directive would always enable the element if the profile is active.
+     */
+    @Input() overwriteDisabled: boolean | null;
```

Comment:
```suggestion
    @Input() overwriteDisabled?: boolean;
```
or
```suggestion
    @Input() overwriteDisabled: boolean | undefined;
```
(not sure which one compiles)

suggestion: Our code style guidelines recommend to avoid `null`.